org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/SnippetCompletionProposal.java

```diff
@@ -310,14 +313,29 @@ private static List<CompletionItem> getGenericSnippets(SnippetCompletionContext
 			final CompletionItem item = new CompletionItem();
 			item.setLabel(template.getName());
 			item.setKind(CompletionItemKind.Snippet);
-			item.setInsertTextFormat(InsertTextFormat.Snippet);
+
 			if (completionItemDefaults.getInsertTextFormat() != null && completionItemDefaults.getInsertTextFormat() == InsertTextFormat.Snippet){
```

Comment:
Do you think if it's necessary to include `isCompletionListItemDefaultsSupport()` in the `if` condition?

Reply:
Yes it should be added